### DISCUSSION TITLE
Latency Interpretation

### DISCUSSION THREAD

**Discussion by chrisui** [RobotWebTools/roslibjs#990] [Q&A]
Just wondering if anyone has cracked a good way to interpret and account for latency with roslibjs and the rosbridge?

I'm attempting to sync web_video_server with data coming through a roslibjs.RosTopic but am unsure how to calculate the latency of the two so data can be buffered and matched appropriately.

**Comment by mvollrath** [comment]
The latency of web_video_server and rosbridge is low and if both are running on the same host the sync should be very tight for visualization purposes.  If rosbridge is lagging behind you are pushing too much data and should use throttling.  If you need the sync to be tighter you should use a native application e.g. rqt.

**Comment by ManuelZ** [comment]
Hi @mvollrath  
- Why could there be some lag between the publisher and the web bridge? 
- What kind of throttling are you referring to? To the parameter of the topic in the roslibjs code?

**Comment by mvollrath** [comment]
> Hi @mvollrath
> 
> * Why could there be some lag between the publisher and the web bridge?

Typically this would be because the publisher and web bridge are on different hosts and have network latency between them.  If they are on the same host and configured properly then no latency should be expected in that link.

> * What kind of throttling are you referring to? To the parameter of the topic in the roslibjs code?

Yes.
